numbat/src/command.rs

```diff
@@ -0,0 +1,436 @@
+use crate::{
+    parser::ParseErrorKind,
+    resolver::{CodeSource, Resolver},
+    span::{SourceCodePositition, Span},
+    ParseError,
+};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum ListItems {
+    Functions,
+    Dimensions,
+    Variables,
+    Units,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Command<'a> {
+    Help,
+    Info { item: &'a str },
+    List { items: Option<ListItems> },
+    Clear,
+    Save { dst: &'a str },
+    Quit,
+}
+
+/// For tracking spans. Contains `(start, start+len)` for each (whitespace-separated)
+/// word in the input
+fn get_word_boundaries(input: &str) -> Vec<(u32, u32)> {
```

Comment:
Could this be implemented using [`str::split_whitespace`](https://doc.rust-lang.org/std/primitive.str.html#method.split_whitespace)?

Reply:
I don't think `split_whitespace` keeps indices, which is what I need. I thought of using `char_indices` and splitting on a predicate, but that would require two `Vec`s which seemed like a needless allocation.